### PR TITLE
task/DES-2419/WG-14: add react leaflet basic map

### PR DESCRIPTION
**Pull request by nathanfranklin**
## Overview: ##

Add react-leaflet map supporting our geoapi map projects that consist of:
- [x] layers
- [x] basic clustered points

Others things needed to fully support our map are not done and now in the following issues: https://jira.tacc.utexas.edu/browse/WG-5 https://jira.tacc.utexas.edu/browse/WG-6

## PR Status: ##

* [x] Ready.

## Related Jira tickets: ##

* was [DES-2419](https://jira.tacc.utexas.edu/browse/DES-2419) but now [WG-14](https://jira.tacc.utexas.edu/browse/WG-14)

## Testing Steps: ##
1. `npm install`   (in README.md under `React client`)
2. `npm run dev` (in README.md under `React client`)
3. go to  `http://localhost:4200/project/2caa321b-3f6a-4faf-b1a7-9717261aec60`

## UI Photos:
Screenshot 2023-03-27 at 4 12 03 PM

## Notes: ##

Using `changey` version of `react-leaflet-markercluster` due to
* https://github.com/yuzhva/react-leaflet-markercluster/issues/187
* https://stackoverflow.com/questions/72062984/react-leaflet-markercluster-is-not-working

### CODE REVIEW

File: react/src/components/Map/Map.tsx [hunk 1]
```diff
@@ -1,7 +1,86 @@
...
+import { TileServerLayer, FeatureCollection } from '../../types';
+import * as L from 'leaflet';
+import { LatLngTuple, MarkerCluster } from 'leaflet';
+import 'leaflet.markercluster';
 
-const Map = () => {
-  return <h2>Map</h2>;
+import 'leaflet/dist/leaflet.css';
+
+/* TODO: review if best approach is to style map with .leaflet-container */
```
Comment (nathanfranklin) [inline react/src/components/Map/Map.tsx:11]: 2 of these todos aren't linked with a jira issue.  i would usually prefer having a jira issue (so there are more notes and a reminder to do it) but maybe it is okay to note some things as todo and not have a jira issue as

File: react/src/components/MapProject/MapProject.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ */
+export default class MapProject extends React.Component<Props> {
+  static defaultProps: Props = {
+    isPublic: false,
+  };
+
+  render() {
+    return (
+      <div style={{ width: '100vw', height: '100vh' }}>
+        {/* TODO for above and project, should we use style-components and/or CSS modules? */}
```
Comment (duckonomy) [inline react/src/components/MapProject/MapProject.tsx:25]: I think this is suffice for the map, but maybe we could decide as we discuss how we'll handle css in the future?

File: react/src/components/Map/Map.tsx [hunk 3]
```diff
@@ -1,7 +1,86 @@
...
+/* TODO: consider createTileLayerComponent
```
Comment (duckonomy) [inline react/src/components/Map/Map.tsx:12]: `createTileLayerComponent` seems nice for situations when you have to update layer props!

**Review by duckonomy — APPROVED** [review]
LGTM! Nice that we have a better organized react way of doing things now 😆

**Pull request merged**